source/llassetgen-cmd/source/main.cpp

```diff
@@ -0,0 +1,226 @@
+
+    // TODO: app.requires_one
+    glyphsOpt->requires_one({fontPathOpt, fontNameOpt});
+
+    // other options
+    unsigned int padding = 0;
+    app.add_option("-p, --padding", padding, paddingHelp);
+
+    unsigned int downsamplingRatio = 1;
+    app.add_option("-w, --downsampling", downsamplingRatio, downsamplingRatioHelp);
+
+    std::string downsampling = "center";
+    app.add_set("--dsalgo", downsampling, algoNames(downsamplingAlgos), downsamplingHelp, true);
+
+    std::vector<int> dynamicRange = {-30, 20};
+    app.add_option("-r, --dynamicrange", dynamicRange, dynamicrangeHelp, true)->requires(distfieldOpt)->expected(2);
+
+    bool createFnt = false;
+    app.add_flag("--fnt", createFnt, fntHelp);
+
+    app.set_config("--config", "", configHelp);
+
+    CLI11_PARSE(app, argc, argv);
+
+    std::string fntPath;
+    std::tie(outPath, fntPath) = outNames(outPath);
+
+    std::set<unsigned long> glyphSet = makeGlyphSet(glyphs, charCodes, includeAscii);
+    if (glyphSet.empty()) {
+        std::cerr << "Error: at least one glyph required" << std::endl;
+        return 2;
+    }
+
+    try {
+        FontFinder fontFinder = fontPathOpt->count() ? FontFinder::fromPath(fontPath)
+                                                     : FontFinder::fromName(fontName);
+
+        std::vector<Image> glyphImages = fontFinder.renderGlyphs(glyphSet, fontSize, padding, downsamplingRatio);
+        std::vector<Vec2<size_t>> imageSizes = sizes(glyphImages, downsamplingRatio);
+        Packing p = packingAlgos[packing](imageSizes.begin(), imageSizes.end(), false);
+
+        if (distfieldOpt->count()) {
+            Image atlas = distanceFieldAtlas(glyphImages.begin(), glyphImages.end(), p, dtAlgos[algorithm],
+                                             downsamplingAlgos[downsampling]);
+            atlas.exportPng<DistanceTransform::OutputType>(outPath, -dynamicRange[0], -dynamicRange[1]);
+        } else {
+            Image atlas = fontAtlas(glyphImages.begin(), glyphImages.end(), p);
+            atlas.exportPng<uint8_t>(outPath);
+        }
+
+        if (createFnt) {
+            std::string faceName = fontNameOpt->count() ? fontName : "Unknown";
+            FntWriter writer{fontFinder.fontFace, faceName, fontSize, 1, false};
+            writer.setAtlasProperties(p.atlasSize, fontSize);
+            writer.readFont(glyphSet.begin(), glyphSet.end());
+            auto gIt = glyphSet.begin();
+            for (auto rectIt = p.rects.begin(); rectIt < p.rects.end(); gIt++, rectIt++) {
+                FT_UInt charIndex = FT_Get_Char_Index(fontFinder.fontFace, static_cast<FT_ULong>(*gIt));
+                writer.setCharInfo(charIndex, *rectIt, {0, 0});
+            }
+            writer.saveFnt(fntPath);
+        }
+    } catch (const std::exception& e) {
+        std::cerr << "Error: " << e.what() << std::endl;
+        return 2;
+    }
+
+    return 0;
+}
+
+int parseDistField(int argc, char** argv) {
```

Comment:
name: why "parse"? you are not really parsing a distance field here. You are calculating a distance field from a given image.

Ah, I think you mean "parse arguments for dist field creation", right? Same above with "parseAtlas".
Mmh... maybe change name to "parseForDistField" / "parseForAtlas". Or maybe some other grammatical sugar.

Reply:
parseDistfieldArgs?

Reply:
sounds good!